### PR TITLE
[TE] frontend - add filter bar to rootcause

### PR DESCRIPTION
**Pull request by ttbach**
- Merged `filter-bar` and `rootcause-filter` components
- Added `filter-bar` to `/rootcause`
new-filter-bar

### CODE REVIEW

File: thirdeye/thirdeye-frontend/app/pods/rootcause/route.js [hunk 1]
```diff
@@ -15,7 +16,8 @@ export default Ember.Route.extend({
       invisibleUrns: new Set(),
       filteredUrns: new Set(),
       hoverUrns: new Set(),
-      context: { urns, anomalyRange, baselineRange, analysisRange }
+      context: { urns, anomalyRange, baselineRange, analysisRange },
+      config
```
Comment (apucher) [inline thirdeye/thirdeye-frontend/app/pods/rootcause/route.js]: for now, please "declare" it as a class-member in controller.js - I'll refactor that shortly and don't want to miss anything

File: thirdeye/thirdeye-frontend/app/mocks/sampleEvents.js [hunk 2]
```diff
@@ -30,8 +30,38 @@ export default [
...
+    displayStart: 1509476399999,
+    displayEnd: 1509479999999,
+    displayLabel: 'Christmas',
+    color: 'green',
+    displayColor: 'green',
+    displayScore: 0.175,
+    humanRelStart: '13 hours after',
+    humanDuration: 'ongoing',
+    "attributes": {
+      "country": ["Canada", "Mexico", "USA"],
```
Comment (apucher) [inline thirdeye/thirdeye-frontend/app/mocks/sampleEvents.js]: you'll have real data shortly. The encoding is "countryCode" -> ["CA", "MX", "US"]

File: thirdeye/thirdeye-frontend/app/pods/components/filter-bar/component.js [hunk 3]
```diff
@@ -86,17 +87,24 @@ export default Ember.Component.extend({
...
         Ember.set(block, 'isHidden', true);
       });
 
       // Note: toggleProperty will not be able to find 'filterBlocks', as it is not an observed property
       // Show clickedBlock
       Ember.set(clickedBlock, 'isHidden', !clickedBlock.isHidden);
-      this.attrs.onTabChange(clickedBlock.eventType);
+
+      const { entities, onSelect } = this.getProperties('entities', 'onSelect');
+      if (onSelect != null) {
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/components/filter-bar/component.js]: nit: `!==` unless you're checking for undefined and false as well, in which case `if (onSelect)` is sufficient

File: thirdeye/thirdeye-frontend/app/pods/components/filter-bar/component.js [hunk 4]
```diff
@@ -86,17 +87,24 @@ export default Ember.Component.extend({
...
+        const urns = Object.keys(entities).filter(urn => entities[urn].type == 'event')
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/components/filter-bar/component.js]: can we just do `entities[urn].type == 'event' && entities[urn].eventType == clickedBlock.eventType` in one pass?

File: thirdeye/thirdeye-frontend/app/mocks/sampleEvents.js (hunk 2 above)
Comment (ttbach) [reply thirdeye/thirdeye-frontend/app/mocks/sampleEvents.js]: Thanks! This was a change I made before you told me about real data, so I'll remove it some later time.

File: thirdeye/thirdeye-frontend/app/pods/components/filter-bar/component.js (hunk 4 above)
Comment (ttbach) [reply thirdeye/thirdeye-frontend/app/pods/components/filter-bar/component.js]: Yep! Thanks for the comment. It's fixed.

File: thirdeye/thirdeye-frontend/app/pods/components/filter-bar/component.js (hunk 3 above)
Comment (ttbach) [reply thirdeye/thirdeye-frontend/app/pods/components/filter-bar/component.js]: Fixed!

File: thirdeye/thirdeye-frontend/app/pods/rootcause/route.js (hunk 1 above)
Comment (ttbach) [reply thirdeye/thirdeye-frontend/app/pods/rootcause/route.js]: Done!

**Pull request merged**